sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/XPathCheck.java

```diff
@@ -79,17 +78,31 @@ private void evaluateXPath() {
 
     try {
       NodeList nodes = (NodeList) getXPathExpressionForDocument(document).evaluate(document, XPathConstants.NODESET);
-      for (int i = 0; i < nodes.getLength(); i++) {
 
+      for (int i = 0; i < nodes.getLength(); i++) {
         int lineNumber = getWebSourceCode().getLineForNode(nodes.item(i));
-        if (message == null) {
-          createViolation(lineNumber, "--");
-        } else {
-          createViolation(lineNumber, message);
+        createViolation(lineNumber);
+      }
+
+    } catch (XPathExpressionException exceptionNodeSet) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1166)